README.md

```diff
@@ -321,6 +321,25 @@ for more information.
 `instance_ip_grouping_key` returns a grouping key with the instance label set
 to the host's IP address.
 
+### Handlers for authentication
+
+If the push gateway you are connecting to is protected with HTTP Basic Auth,
+you can use a special handler to set the Authorization header.
+
+```python
+from prometheus_client import CollectorRegistry, Gauge, push_to_gateway
+from prometheus_client.handlers.basic_auth import handler as basic_auth_handler
+import os
+
+def my_auth_handler(url, method, timeout, headers, data):
+    username = os.environ['PUSHGW_USERNAME']
+    password = os.environ['PUSHGW_PASSWORD']
+    return basic_auth_handler(url, method, timeout, headers, data, username, password)
```

Comment:
I'd hardcode the user/pw here to keep in simpler.

If you use an env example, some users will be confused and think it only works if it comes from the env.